v6_MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,226 @@
+# Braintree Android Drop-In v6 Migration Guide
+
+See the [CHANGELOG](/CHANGELOG.md) for a complete list of changes. This migration guide outlines the basics for updating your Braintree Drop-In integration from v5 to v6.
+
+v6 of the Drop-In SDK is compatible with v4 of the [Braintree Android SDK](https://github.com/braintree/braintree_android).
+
+_Documentation for v6 will be published to https://developer.paypal.com/braintree/docs once it is available for general release._
+
+## Table of Contents
+
+1. [Gradle](#gradle)
+1. [Builder Pattern](#builder-pattern)
+1. [DropInRequest](#dropinrequest)
+1. [Authorization](#authorization)
+1. [Launch Drop-In](#launch-drop-in)
+1. [Handle Drop-In Result](#handle-drop-in-result)
+1. [Fetch Last Used Payment Method](#fetch-last-used-payment-method)
+
+## Gradle
+
+Add the dependency in your `build.gradle`:
+
+```groovy
+dependencies {
+  implementation 'com.braintreepayments.api:drop-in:6.0.0'
+}
+```
+
+The credentials for accessing the Cardinal Mobile SDK have changed.
+
+Update the credentials in your top-level build.gradle to the following:
+
+```groovy
+repositories {
+    maven {
+        url "https://cardinalcommerceprod.jfrog.io/artifactory/android"
+        credentials {
+            username 'braintree_team_sdk'
+            password 'AKCp8jQcoDy2hxSWhDAUQKXLDPDx6NYRkqrgFLRc3qDrayg6rrCbJpsKKyMwaykVL8FWusJpp'
+        }
+    }
+}
+```
+
+## Builder Pattern
+
+The builder pattern has been removed in v6 to allow for consistent object creation across Java and Kotlin. 
+Method chaining has been removed, and setters have been renamed with the `set` prefix.
+
+For example, a `DropInRequest` can now be constructed below:
+
+Java:
+```java
+DropInRequest request = new DropInRequest();
+request.setCollectDeviceData(true);
+request.setMaskCardNumber(true);
+```
+
+Kotlin:
```

Comment:
I added both Kotlin and Java snippets here to make it easier to write developer docs for this version, but they could be removed since we don't have Kotlin snippets in the Android core v4 migration guide.

Reply:
Great idea! I like having both.